core/src/main/java/hudson/security/AccessDeniedException2.java

```diff
@@ -38,8 +41,14 @@ public AccessDeniedException2(Throwable t, Authentication authentication, Permis
      */
     public void reportAsHeaders(HttpServletResponse rsp) {
         rsp.addHeader("X-You-Are-Authenticated-As",authentication.getName());
-        for (GrantedAuthority auth : authentication.getAuthorities()) {
-            rsp.addHeader("X-You-Are-In-Group",auth.getAuthority());
+        GrantedAuthority[] authorities = authentication.getAuthorities();
```

Comment:
do we even need to report this given there is a `whoAmI` which reports this in a much nicer way (and most users won't check the headers, and most scripts can't take any action)